src/orderbook.ts

```diff
@@ -0,0 +1,34 @@
+export class Orderbook {
+  from: string;
+  to: string;
+  asks: Map<number, number>;
+  bids: Map<number, number>;
+
+  constructor(from: string, to: string) {
+    this.from = from;
+    this.to = to;
+    this.asks = new Map();
+    this.bids = new Map();
+  }
+
+  addBid(price: number, volume: number) {
```

Comment:
Should the price be numerator and denominator? Can we just represent the price with one number?

Reply:
At some point we should map the price to a single number as it makes practical calculation much simpler. We can leave it as a fraction internal this class in order to avoid rounding errors e.g. when inverting order books.